src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -145,5 +173,32 @@ private IFeatureFilter GetFeatureFilter(string filterName)
 
             return filter;
         }
+
+        private ContextualFeatureFilterEvaluator GetContextualFeatureFilter(string filterName, Type appContextType)
+        {
+            if (appContextType == null)
+            {
+                throw new ArgumentNullException(nameof(appContextType));
+            }
+
+            ContextualFeatureFilterEvaluator filter = _contextualFeatureFilterCache.GetOrAdd(
+                $"{filterName}{Environment.NewLine}{appContextType.FullName}",
+                (_) => {
+
+                    IFeatureFilterMetadata metadata = GetFeatureFilterMetadata(filterName);
+
+                    ContextualFeatureFilterEvaluator f = null;
+
+                    if (ContextualFeatureFilterEvaluator.GetContextualFilterInterface(metadata, appContextType) != null)
+                    {
+                        f = new ContextualFeatureFilterEvaluator(metadata, appContextType);
+                    }
+
+                    return f;
```

Comment:
Can we hide this logic inside ContextualFeatureFilterEvaluator class? It might be a little too much to know the details from outside. Make it like below with a static function

return ContextualFeatureFilterEvaluator.Create(metadata, appContextType);

Reply:
How about

```
  return ContextualFeatureFilterEvaluator.IsContextualFilter(metadata, appContextType) ?
    new ContextualFeatureFilterEvaluator(metadata, appContextType) :
    null;
```

---
In reply to: [328393965](https://github.com/microsoft/FeatureManagement-Dotnet/pull/14#discussion_r328393965) [](ancestors = 328393965)